sdk/src/main/java/com/mapbox/maps/renderer/FpsManager.kt

```diff
@@ -0,0 +1,186 @@
+      val frameRenderTimeNs = System.nanoTime() - preRenderTimeNs
+      frameRenderTimeAccumulatedNs += frameRenderTimeNs
+      // normally we update FPS counter once a second
+      if (choreographerTicks >= screenRefreshRate) {
+        calculateFps(listener)
+      } else {
+        // however to produce correct values we also update FPS after IDLE_TIMEOUT_MS
+        // otherwise when updating the map after it was IDLE first update will report
+        // huge delta between new frame and last frame (as we're using dirty rendering)
+        handler.postDelayed(IDLE_TIMEOUT_MS, fpsManagerToken) {
+          calculateFpsAndReset(listener)
+        }
+      }
+      preRenderTimeNs = -1L
+    }
+  }
+
+  fun onSurfaceDestroyed() {
+    fpsChangedListener?.let { listener ->
+      calculateFpsAndReset(listener)
+    }
+  }
+
+  fun destroy() {
+    handler.removeCallbacksAndMessages(fpsManagerToken)
+    fpsChangedListener = null
+  }
+
+  private fun calculateFpsAndReset(listener: OnFpsChangedListener) {
+    handler.removeCallbacksAndMessages(fpsManagerToken)
+    calculateFps(listener)
+    previousRefreshNs = -1
+  }
+
+  private fun performPacing(): Boolean {
+    val drawnFrameIndex = (choreographerTicks * userDefinedRatio).toInt()
+    if (LOG_STATISTICS) {
+      logI(
+        TAG,
+        "Performing pacing, current index=$drawnFrameIndex," +
+          " previous drawn=$previousDrawnFrameIndex, proceed with rendering=${drawnFrameIndex > previousDrawnFrameIndex}"
+      )
+    }
+    if (drawnFrameIndex > previousDrawnFrameIndex) {
+      previousDrawnFrameIndex = drawnFrameIndex
+      return true
+    }
+    choreographerSkips++
+    return false
+  }
+
+  private fun calculateFps(listener: OnFpsChangedListener) {
+    if (choreographerTicks == 0) {
+      return
+    }
+    val droppedFps = choreographerSkips.toDouble() / choreographerTicks
+    val fps = (1.0 - droppedFps) * screenRefreshRate
+    val averageRenderTimeNs = frameRenderTimeAccumulatedNs.toDouble() / choreographerTicks
+    listener.onFpsChanged(fps)
+    if (LOG_STATISTICS) {
+      logW(
+        TAG,
+        "VSYNC based FPS is $fps," +
+          " average core rendering time is ${averageRenderTimeNs / 10.0.pow(6.0)} ms," +
+          " missed $choreographerSkips out of $choreographerTicks VSYNC pulses"
+      )
+    }
+    previousDrawnFrameIndex = 0
+    frameRenderTimeAccumulatedNs = 0L
+    choreographerTicks = 0
+    choreographerSkips = 0
+  }
+
+  private companion object {
+    private const val TAG = "Mbgl-FpsManager"
+    private const val USER_DEFINED_REFRESH_RATE_NOT_SET = -1
+    private const val USER_DEFINED_RATIO_NOT_SET = 0.0
+    private const val SCREEN_METRICS_NOT_DEFINED = -1
+    private const val LOG_STATISTICS = true
+    private const val IDLE_TIMEOUT_MS = 50L
+
+    /**
+     * We use same handler and looper as our render thread and we need to make sure that
+     * we clear messages related to FPS manager only.
+     */
```

Comment:
Huh?

Reply:
I'll re-phrase a bit

Reply:
just removed, actually it's pretty clear why it is needed